docs/src/main/sphinx/security/user-mapping.rst

```diff
@@ -2,67 +2,123 @@
+one authentication method can have the same authentication type. For example,
+the Password file, LDAP, and Salesforce authentication methods all share the
+``PASSWORD`` authentication type.
+
+You can specify different user mapping patterns for different authentication
+types when multiple authentication methods are enabled:
 
 ===================================== ===============================================================
-Authentication                        Property
+Authentication type                   Property
 ===================================== ===============================================================
-Username and Password (file or LDAP)  ``http-server.authentication.password.user-mapping.pattern``
-Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
+Password (file, LDAP, Salesforce)     ``http-server.authentication.password.user-mapping.pattern``
+OAuth2                                ``http-server.authentication.oauth2.user-mapping.pattern``
 Certificate                           ``http-server.authentication.certificate.user-mapping.pattern``
-Json Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+JSON Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
 Insecure                              ``http-server.authentication.insecure.user-mapping.pattern``
 ===================================== ===============================================================
 
+.. _pattern-file:
+
 File mapping rules
 ------------------
 
-The file mapping rules allow for more complex mappings from the authentication user.  These rules are
-loaded from a JSON file defined in a configuration property.  The mapping is based on the first matching
-rule, processed from top to bottom. If no rules match, authentication is denied.  Each rule is composed of the
+Use the **File mapping rules** method if your authentication provider’s way to
+express usernames cannot be reduced to a single rule, or if you want to exclude
+a set of users from accessing the cluster.
+
+The rules are loaded from a JSON file identified in a configuration property.
+The mapping is based on the first matching rule, processed from top to bottom.
+If no rules match, authentication is denied.  Each rule is composed of the
 following fields:
 
-* ``pattern`` (required): regex to match against authentication user.
-* ``user`` (optional): replacement string to substitute against pattern.
+* ``pattern`` (required): regex to match against the authentication method's
+  username.
+* ``user`` (optional): replacement string to substitute against *pattern*.
   The default value is ``$1``.
-* ``allow`` (optional): boolean indicating if the authentication should be allowed.
+* ``allow`` (optional): Boolean indicating whether authentication is to be
```

Comment:
Why is boolean capitalized here but the other items are lowercase?

Reply:
I always learned that Boolean was init-capped because a mathematician named George Boole came up with Boolean logic. See https://en.wikipedia.org/wiki/Boolean_data_type. But I'll lowercase it here for fellowship with the other words.

Reply:
I think boolean has become a normal adjective by now .. although Meriam Webster disagrees with @dain and myself.. https://www.merriam-webster.com/dictionary/Boolean

Reply:
In java and other languages, Boolean is capped if is an object and lower case if it is instead a primitive. I would treat this the same as if using the word "integer" in its normal dictionary context here so as to keep with the idea that we are dealing with a primitive.

Reply:
Agreed as usual with m57lyra except that Java is init-capped when referring to the language and only lowercased to refer to the command. 
(Running, ducking, looking for armor ...)

Reply:
FAIR.